### PR TITLE
Update Pokémon Service to use Smithy IDL v2 (#2523)

### PR DESCRIPTION
**Pull request by OBBO67**
## Motivation and Context
Smithy IDL v2 has been released. The example Pokemon Service model should be updated to use v2.

[Open issue 2523](https://github.com/awslabs/smithy-rs/issues/2523).

## Description
Made appropriate changes to pokemon.smithy following the [Smithy IDL v1 -> v2 migration guide](https://smithy.io/2.0/guides/migrating-idl-1-to-2.html).

The `StreamingBlob` was not marked as `@required` so I made it optional in the `StreamPokemonRadioOutput`.

**Additional Changes:**
* Fix model link in Pokemon example make file.
* Add `*.iml` files to `.gitignore`.

## Testing
Manual test by running the example Pokemon Service

## Checklist
N/A

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,6 +9,9 @@
 # BlueJ files
 *.ctxt
 
+# IntelliJ module file
+*.iml
+
```
Comment (82marbag) [inline .gitignore:14]:
```suggestion
```

See the comment at the top of the file

Comment (OBBO67) [reply .gitignore:14]: my bad, cheers

**Review by david-perez — COMMENTED** [review]
This is a welcome patch, but in my opinion does not close the original #2523 issue. Smithy IDL v2 is _mostly_ backwards-compatible with v1, as the small diff in this PR shows, but IDL v2 has several interesting features / syntax additions that we should showcase in our examples. A small list can be found under the optional migration steps section: https://smithy.io/2.0/guides/migrating-idl-1-to-2.html#optional-migration-steps

**Comment by david-perez** [comment]
Closing this in favor of #3494.

**Pull request closed without merging**